python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,105 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+    """
+
+    def __init__(
+        self, model: torch.nn.Module, preprocessor: Optional[Preprocessor] = None
+    ):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model: If the checkpoint contains a model state dict, and not
+                the model itself, then the state dict will be loaded to this
+                ``model``.
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        feature_columns: Optional[
+            Union[List[str], List[List[str]], List[int], List[List[int]]]
+        ] = None,
+        dtype: Optional[torch.dtype] = None,
+    ) -> DataBatchType:
+        """Run inference on data batch.
+
+        The data is converted into a torch Tensor before being inputted to
+        the model.
```

Comment:
So we are not supporting everything that's supported by `Dataset.to_torch`? I think it would make sense to reuse the logic and arguments.

Reply:
The arguments that are not being included are either related to label columns (which is not relevant for inference) or specific to datasets (batch size, prefetch_blocks, etc.).